File: api/src/main/java/net/blueshell/api/dto/SocialDTO.java
package net.blueshell.api.dto;


import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import lombok.EqualsAndHashCode;
import net.blueshell.api.base.BaseDTO;
import net.blueshell.api.common.enums.PlatformType;

@Data
@EqualsAndHashCode(onlyExplicitlyIncluded = true, callSuper = true)
@Schema(name = "Social")
public class SocialDTO extends BaseDTO {
    private String title;
    private String text;
    private String url;
    private PlatformType[] platforms;
}

File: api/src/main/kotlin/net/blueshell/api/dto/SocialDTO.kt
package net.blueshell.api.dto

import io.swagger.v3.oas.annotations.media.Schema
import net.blueshell.api.base.dto.BaseDTO
import net.blueshell.api.common.enums.PlatformType

@Schema(name = "Social")
data class SocialDTO(
    var title: String? = null,
    var text: String? = null,
    var url: String? = null,
    var platforms: Array<PlatformType> = emptyArray(),
) : BaseDTO()
